src/playlist-maker.js

```diff
@@ -248,21 +245,27 @@ export default function factory(player, initialList, initialIndex = 0) {
 
       // @todo - Simplify this to `list.slice()` for v5.
       const previousPlaylist = Array.isArray(list) ? list.slice() : null;
+      const nextPlaylist = newList.slice();
```

Comment:
We are double-`slice`ing here because below we want to pass `nextPlaylist` through the `duringplaylistchange` event. Previously, this was leaking the internal playlist abstractions during this event.